| a. ¿Qué representa una fila del dataset?                                  | la información de cada estudiante que realizo la encuesta                             |
|---------------------------------------------------------------------------|---------------------------------------------------------------------------------------|
| b. ¿Cuál es la unidad de análisis?                                        | son los estudiantes de la universidad de estudio                                      |
| c. ¿Cuántas observaciones y variables hay?                                | En total encontramos 2512 observaciones y 19 variables que se representan en la tabla |
| d. ¿Qué parece representar el dataset en términos del problema o dominio? |   Representa la vida academica y personal de los estudiantes durante el mes de agosto, lo que sugiere que busca relacionar los habitos con la vida academica.                                                                                  |

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv("estudiantes - copy.csv")

In [ ]:
df.head()

,student_id,programa,semestre,edad,genero,horas_estudio_semana,horas_sueno_noche,tiempo_desplazamiento_min,medio_transporte,nivel_estres,satisfaccion_academica,dias_ejercicio_semana,horas_sociales_semana,uso_ia_generativa,confianza_programando,area_interes_datos,modalidad_preferida,promedio_acumulado,fecha_encuesta
0,E0001,Economía,5,21,Mujer,16.8,5.5,61.0,Bus,4,4,3,7.2,A veces,3,Ingeniería de datos,Presencial,3.47,2026-08-01
1,E0002,Administración,6,23,Mujer,23.5,7.2,98.0,TransMilenio,4,4,2,10.8,Nunca,2,No estoy seguro,Presencial,3.69,2026-08-10
2,E0003,Administración,1,17,Hombre,12.6,6.7,55.0,TransMilenio,5,4,3,6.8,A veces,4,Analítica de negocio,Presencial,3.26,2026-08-21
3,E0004,Ciencia de Datos,8,21,Mujer,15.7,6.1,57.0,Moto,4,3,1,14.3,Nunca,4,Analítica de negocio,Presencial,2.90,2026-08-19
4,E0005,Diseño,3,22,Mujer,20.5,6.4,50.0,Carro,5,4,5,11.0,A veces,2,No estoy seguro,Presencial,3.54,2026-08-04


In [ ]:
df.tail()

,student_id,programa,semestre,edad,genero,horas_estudio_semana,horas_sueno_noche,tiempo_desplazamiento_min,medio_transporte,nivel_estres,satisfaccion_academica,dias_ejercicio_semana,horas_sociales_semana,uso_ia_generativa,confianza_programando,area_interes_datos,modalidad_preferida,promedio_acumulado,fecha_encuesta
2507,E2113,Ciencia de Datos,1,18,Mujer,13.7,6.3,81.0,Caminando,5,4,5,9.7,Frecuentemente,3,Analítica de negocio,Presencial,3.67,2026-08-10
2508,E0456,Administración,2,19,Mujer,17.7,5.8,47.0,Caminando,4,4,3,4.9,Frecuentemente,1,IA y ML,Presencial,3.24,2026-08-04
2509,E1001,Ingeniería de Sistemas,1,19,Mujer,20.1,5.2,92.0,Caminando,5,3,1,9.1,Frecuentemente,4,Analítica de negocio,Presencial,3.82,2026-08-19
2510,E0438,Ingeniería Industrial,1,19,Mujer,15.5,7.8,50.0,TransMilenio,3,4,5,3.7,Todos los días,3,Analítica de negocio,Presencial,3.79,2026-08-11
2511,E1192,Administración,2,20,Mujer,15.6,7.7,50.0,Caminando,3,4,6,10.8,A veces,4,Visualización,Presencial,3.75,2026-08-02


In [ ]:
df.shape


(2512, 19)

In [ ]:
df.columns

Index(['variables', 'tipo_python', 'escala', 'descripcion', 'ejemplo'], dtype='str')

In [ ]:
tipo = df.dtypes
print(tipo)

student_id                       str
programa                         str
semestre                       int64
edad                           int64
genero                           str
horas_estudio_semana         float64
horas_sueno_noche            float64
tiempo_desplazamiento_min    float64
medio_transporte                 str
nivel_estres                   int64
satisfaccion_academica         int64
dias_ejercicio_semana          int64
horas_sociales_semana        float64
uso_ia_generativa                str
confianza_programando          int64
area_interes_datos               str
modalidad_preferida              str
promedio_acumulado           float64
fecha_encuesta                   str
dtype: object


In [ ]:

escala = ["ordinal","Nominal","continua","continua","nominal","continua","continua","continua","nominal","ordinal","ordinal","discreto","discreto","ordinal","ordinal","nominal","ordinal","discreto","nominal"]
tipos = ["str","str","int64","int64","str","float64","float64","float64","str","int64","int64","int64","float64","str","int64","str","str","float64","str"]
ejemplo = ["E0456","Administración","2","19","Mujer","17.7","5.8","47.0","Caminando","4","4","3","4.9","Frecuentemente","1","IA y ML","Presencial","3.24","2026-08-04"]
descripcion = ["es el identificador unico de cada estudiante de la universidad","la carrera que esta cursando","cuantos periodos de estudio lleva el estudiante en la institucion","cuantos años tiene el estudiante","con que sexo se identifica","tiempo que dedico a sus actividades academicas","tiempo que dedico a dormir en las horas de la noche","cuanto tiempo tardo llendo de su casa a la universidad o viceversa","tipo de vehiculo o servicio que uso para llegar","como calificaria su carga de materias","que tan bien sen sienten con su situacion academica","cuantos dias hacen actividad fisica en la semana","cuanto tiempo conviven con otras personas en la semana","que tanto usan la ia en sus vidad","que tambien creen que pueden programar","materia en la que quisieran ahondar mas adelante","como estan tomando clases","cual es la nota central segun todas las materias que este cursando","dia mes y año que fue hecha la encuesta"]
dcd = {
       "variables" : df.columns,
       "tipo_python":tipos,
       "escala": escala,
       "descripcion": descripcion,
       "ejemplo": ejemplo}

ef = pd.DataFrame(dcd, index = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19] )
ef

,variables,tipo_python,escala,descripcion,ejemplo
1,student_id,str,ordinal,es el identificador unico de cada estudiante d...,E0456
2,programa,str,Nominal,la carrera que esta cursando,Administración
3,semestre,int64,continua,cuantos periodos de estudio lleva el estudiant...,2
4,edad,int64,continua,cuantos años tiene el estudiante,19
5,genero,str,nominal,con que sexo se identifica,Mujer
6,horas_estudio_semana,float64,continua,tiempo que dedico a sus actividades academicas,17.7
7,horas_sueno_noche,float64,continua,tiempo que dedico a dormir en las horas de la ...,5.8
8,tiempo_desplazamiento_min,float64,continua,cuanto tiempo tardo llendo de su casa a la uni...,47.0
9,medio_transporte,str,nominal,tipo de vehiculo o servicio que uso para llegar,Caminando
10,nivel_estres,int64,ordinal,como calificaria su carga de materias,4


In [ ]:
df.isnull().sum()

student_id                    0
programa                      0
semestre                      0
edad                          0
genero                        0
horas_estudio_semana          0
horas_sueno_noche            72
tiempo_desplazamiento_min    56
medio_transporte              0
nivel_estres                  0
satisfaccion_academica        0
dias_ejercicio_semana         0
horas_sociales_semana         0
uso_ia_generativa             0
confianza_programando         0
area_interes_datos           41
modalidad_preferida           0
promedio_acumulado            0
fecha_encuesta                0
dtype: int64

In [ ]:
df.nunique()

student_id                   2500
programa                        8
semestre                       10
edad                           13
genero                          5
horas_estudio_semana          305
horas_sueno_noche              57
tiempo_desplazamiento_min     135
medio_transporte                7
nivel_estres                    5
satisfaccion_academica          4
dias_ejercicio_semana           8
horas_sociales_semana         189
uso_ia_generativa               4
confianza_programando           5
area_interes_datos              6
modalidad_preferida             3
promedio_acumulado            191
fecha_encuesta                 21
dtype: int64

In [ ]:
variables = df.columns
i = 0
while i < len(variables):
    var = variables[i]
    print( df[var].duplicated().sum() )
    i += 1

12
2504
2502
2499
2507
2207
2454
2376
2505
2507
2508
2504
2323
2508
2507
2505
2509
2321
2491


| Aspectos     | Pregunta guia                                         | Respuesta                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
|--------------|-------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Dimensiones  | ¿Qué tan grande es el dataset?                        | Todos tienen sentido, algunos son valores numericos o str sin embargo representan una variale de escala ordinal, y la fecha ya que deberia de usarse como un tipo datetime64                                                                                                                                                                                                                                                                                                                                     |
| Tipos        | ¿Los tipos inferidos tienen sentido?                  | la tabla tiene 2512 observaciones y 19 variables de estudio                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
| Faltantes    | ¿Dónde se concentran los valores ausentes?            | los valores faltantes se encuentran distribuidos de la siguiente manera: horas_sueno_noche = 72  tiempo_desplazamiento_min = 56 area_interes_datos = 41 para un total de 169 faltantes.                                                                                                                                                                                                                                                                                                                          |
| Duplicados   | ¿Existen filas repetidas?                             | la cantidad de valores unicos de cada variable son:  student_id = 12 programa 2504 semestre = 2502 edad = 2499 genero = 2507 horas_estudio_semana = 2207 horas_sueno_noche = 2454 tiempo_desplazamiento_min = 2376 medio_transporte = 2505 nivel_estres = 2507 satisfaccion_academica = 2508 dias_ejercicio_semana = 2504 horas_sociales_semana = 2323 uso_ia_generativa = 2508 confianza_programando = 2507 area_interes_datos = 2505 modalidad_preferida = 2509 promedio_acumulado = 2321 fecha_encuesta= 2491 |
| Cardinalidad | ¿Cuántos valores distintos tiene cada variable?       | los valores unicos de cada variable son:  student_id = 2500  programa  8 semestre = 10 edad = 13 genero = 5 horas_estudio_semana = 305 horas_sueno_noche = 57 tiempo_desplazamiento_min = 135 medio_transporte = 7 nivel_estres = 5 satisfaccion_academica = 4 dias_ejercicio_semana = 8 horas_sociales_semana = 189 uso_ia_generativa = 4 confianza_programando = 5 area_interes_datos = 6 modalidad_preferida = 3 promedio_acumulado = 191 fecha_encuesta= 21                                                  |
| Distribución | ¿Qué rangos, categorías o valores llaman la atencion? |   En hora_estudio_semana hay valores sospechosamente muy altos como 95 horas y en Programa EXISTEN dos variables que significan lo mismo pero escritas de diferente manera"Ciencia Datos" y "Ciencia de Datos"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |


| pregunta                                                                               | Respuesta                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
|----------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ¿Hay columnas cuyo significado o nombre no sea suficientemente claro?                  | el tiempo de desplazamiento ya que no conocemos si es a la entrada o salida y como esto puede llegar a afectar. satisfacción académica ya que no indica si es en cuanto a las materias notas o profesores.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
| ¿Existen códigos, categorías o unidades que requerirían documentación adicional?       | no, las categorías se entienden bien, aunque las escalas del 1 al 5 (estrés, satisfacción, confianza) no explican qué significa cada número                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
| ¿Hay variables cuyo tipo en Python no corresponda con su significado analítico?        | el único error es con respecto a las fechas ya que estan como un str y deberian ser datetime64                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
| ¿Detectan categorías inconsistentes, rangos improbables o posibles errores de captura? |  Existen categorias duplicadas  en programa ("Ciencia Datos" y "Ciencia de Datos") , y valores muy exagerados en hora_estudio_semana (95hrs)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
| ¿Qué información adicional pedirían al dueño de los datos antes de analizar?           | el lugar y hora de salida hacia la universidad para conocer desde donde se demoran más en llegar y ver como eso se relaciona con las horas de sueño  y la satisfacción académica y el nivel de estrés. Estado de ánimo general para conocer cómo se sienten durante el semestre y así armar los cursos a dictar. tipo de uso de ia de escala ordinal y relacionarlo con el promedio general para saber si en un uso inadecuado cambia drásticamente con respecto a uso menor Cantidad de amigos cercanos ya que esto junto con las horas de socialización y el tipo de uso de ia junto con el estado de ánimo general nos mostraría que tipo de personas son más propensas a tener una insatisfacción de la experiencia universitaria |
| ¿Qué riesgos tendría comenzar un análisis sin resolver estas dudas?                    | al momento de desarrollar estrategias para mejorar la experienca educativa podria tener un enfoque que difiere del contexto especifico que viven los estudiantes llendo a un campo muy general y que se vuelve poco efectivo a la hora de mitigar problemas                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |

| Pregunta analitica | variables necesarias | ¿Datos suficientes? ¿Por qué? |
|---|---|---|
| ¿Como afecta el lugar de residencia en el desempeño academico de los estudiantes? | desempeño academico, dias de ejercisio, horas de sueño, tiempo de desplazamiento, id de estudiante, programa que esta cursando,lugar de residencia, semestre en el que esta, horas de socializacion | no, ya que falta el lugar de residencia para conocer el lugar de residencia ya que este afecta variables como el tiempo de desplazamiento entre otras. |
| ¿El tiempo de desplazamiento y las horas de sueño afectan el nivel de estrés y la satisfacción académica? | tiempo de desplazamiento, horas de sueño, nivel de estres, satisfaccion academica, medio de transporte | parcialmente, ya que nivel de estres y satisfaccion academica estan completas, pero tiempo de desplazamiento (56 datos faltantes) y horas de sueño (72 datos faltantes) tienen valores nulos que tocaria tratar antes, ademas hay valores muy altos (hasta 360 minutos) que parecen errores de captura |
| ¿El uso de inteligencia artificial generativa se relaciona con la confianza al programar y el promedio academico? | uso de ia generativa, confianza programando, promedio acumulado, area de interes en datos, programa que esta cursando | si, ya que uso de ia generativa, confianza programando y promedio acumulado no tienen datos faltantes, aunque area de interes en datos tiene 41 datos faltantes que reducirian un poco la muestra si se cruza por esa variable |

¿cómo viven los estudiantes su experiencia educativa?, Desde la dirección de bienestar de la universidad se quiere encontrar que factores afectan el desempeño académico y desarrollar programas de acompañamiento para los estudiantes por esto hicieron una encuesta en donde recogieron la siguiente información:  calidad del sueño, niveles de
estrés, modalidad de estudio, entre otras
y percepción general de la vida universitaria.
Después de revisar la base de datos entregada nos damos cuenta de que tenemos datos muy generales y que no toman en cuenta muchos factores como el lugar de residencia o la cantidad de amigos cosas que afectan la experiencia universitaria, por lo tanto, aunque se puede utilizar el dataset, se queda corto al momento de querer identificar el principal problema en un contexto determinado.
Lo que deberíamos analizar es cómo se comporta las horas de sueño, tiempo de trasporte, horas sociales veces de hacer deporte a la semana y uso de ia con el desempeño académico y cómo se comporta con los cambios que presenta cada uno.
Por último, lo siguiente que tocaría hacer es recolectar los datos faltantes y ver como estos se relación con los que ya tenemos para ver si esto afecta directa o indirectamente el desempeño académico y poder tener un contexto mas amplio de lo que pasa con los diferentes tipos de estudiantes que hay en la universidad para tener un impacto positivo y duradero con los programas de acompañamiento 

